source/docs/software/commandbased/organizing-command-based.rst

```diff
@@ -0,0 +1,259 @@
+
+What Bad Organization Looks Like
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Poorly organized code often has internal structure that makes little to no sense, even to whoever wrote it. When functionality has to be added or changed, it often breaks unrelated parts of the robot: adding automatic shooter control might introduce a bug in the climbing sequence for unclear reasons. Alternatively, the organizational framework might be so strict that it's impossible to implement necessary behavior, requiring nasty hacks or workarounds. Many lines of boilerplate code are needed for simple robot logic. Collaboration among multiple programmers is difficult or impossible.
+
+Defining Commands
+-----------------
+
+In larger robot codebases, multiple copies of the same command need to be used in many different places. For instance, a command that runs a robot's intake might be used in teleop, bound to a certain button; as part of a complicated command group for an autonomous routine; and as part of a self-test sequence.
+
+As an example, let's look at some ways to define a simple command that simply runs the robot's intake forward at full power until cancelled.
+
+Inline Commands
+^^^^^^^^^^^^^^^
+
+The easiest and most expressive way to do this is with a ``StartEndCommand``:
+
+.. tabs::
+
+  .. code-tab:: java
+
+    Command runIntake = new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake);
+
+  .. code-tab:: c++
+
+    // TODO
+
+This is sufficient for commands that are only used once. However, for a command like this that might get used in many different autonomous routines and button bindings, inline commands everywhere means a lot of repetitive code:
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // RobotContainer.java
+    intakeButton.whileHeld(new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake));
+
+
+    Command intakeAndShoot = new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake)
+        .alongWith(new RunShooter(shooter));
+
+    Command autonomousCommand = new ParallelCommandGroup(
+        new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake).withTimeout(5),
+        new WaitCommand(3),
+        new StartEndCommand(() -> intake.set(1), () -> intake.set(0), intake).withTimeout(5)
+    )
+
+  .. code-tab:: c++
+
+    // TODO
+
+(Creating one ``StartEndCommand`` instance and putting it in a variable won't work here, since once an instance of a command is added to a command group it is effectively "owned" by that command group and cannot be used in any other context.)
```

Comment:
I'd have this as a regular paragraph or in an admonition instead of in parentheses.